ee/server/service/software_installers.go

```diff
@@ -80,12 +86,26 @@ func (svc *Service) UploadSoftwareInstaller(ctx context.Context, payload *fleet.
 
 	// TODO: QA what breaks when you have a software title with no versions?
 
+	var teamName string
+	if payload.TeamID != nil {
+		t, err := svc.ds.Team(ctx, *payload.TeamID)
+		if err != nil {
+			return err
+		}
+		teamName = t.Name
+	}
+
+	// Create activity
+	if err := svc.ds.NewActivity(ctx, vc.User, fleet.ActivityTypeAddedSoftware{SoftwareTitle: title, SoftwarePackage: payload.Filename, TeamName: teamName}); err != nil {
+		return nil
```

Comment:
Return the error wrapped in context here, e.g. `return ctxerr.Wrap(ctx, err, "creating activity for added software")`.

Reply:
Derp not sure why I left this as just `nil`... brain fart! good catch!